kafka/channel/pkg/dispatcher/dispatcher.go

```diff
@@ -63,6 +61,64 @@ type KafkaDispatcher struct {
 	logger    *zap.Logger
 }
 
+func NewDispatcher(ctx context.Context, args *KafkaDispatcherArgs) (*KafkaDispatcher, error) {
+	conf := sarama.NewConfig()
+	conf.Version = sarama.V2_0_0_0
+	conf.ClientID = args.ClientID
+	conf.Consumer.Return.Errors = true // Returns the errors in ConsumerGroup#Errors() https://godoc.org/github.com/Shopify/sarama#ConsumerGroup
+	client, err := sarama.NewClient(args.Brokers, conf)
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to create kafka client: %v", err)
+	}
+
+	producer, err := sarama.NewAsyncProducerFromClient(client)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create kafka producer: %v", err)
+	}
+
+	messageSender, err := kncloudevents.NewHttpMessageSender(args.KnCEConnectionArgs, "")
+	if err != nil {
+		args.Logger.Fatal("failed to create message sender", zap.Error(err))
+	}
+
+	dispatcher := &KafkaDispatcher{
+		dispatcher:           eventingchannels.NewMessageDispatcher(args.Logger),
+		kafkaConsumerFactory: kafka.NewConsumerGroupFactory(client),
+		kafkaConsumerGroups:  make(map[eventingchannels.ChannelReference]map[subscription]sarama.ConsumerGroup),
+		kafkaAsyncProducer:   producer,
+		logger:               args.Logger,
+		messageSender:        messageSender,
+		handler:              args.Handler,
+		topicFunc:            args.TopicFunc,
+	}
+	receiverFunc, err := eventingchannels.NewMessageReceiver(
+		func(ctx context.Context, channel eventingchannels.ChannelReference, message binding.Message, transformers []binding.TransformerFactory, _ nethttp.Header) error {
+			kafkaProducerMessage := sarama.ProducerMessage{
+				Topic: args.TopicFunc(utils.KafkaChannelSeparator, channel.Namespace, channel.Name),
+			}
+
+			err := protocolkafka.WriteProducerMessage(ctx, message, &kafkaProducerMessage, transformers...)
+			if err != nil {
+				return err
+			}
+
+			dispatcher.kafkaAsyncProducer.Input() <- &kafkaProducerMessage
+			return nil
+		},
+		args.Logger,
+		eventingchannels.ResolveMessageChannelFromHostHeader(dispatcher.getChannelReferenceFromHost))
+	if err != nil {
+		return nil, err
+	}
+
+	dispatcher.receiver = receiverFunc
+	dispatcher.setConfig(&multichannelfanout.Config{})
+	dispatcher.setHostToChannelMap(map[string]eventingchannels.ChannelReference{})
+	dispatcher.topicFunc = args.TopicFunc
```

Comment:
Do we need to set this again? will it change from the initial assignment on https://github.com/knative/eventing-contrib/pull/1039/files#diff-729c8026514d13072e43e868cd84cae5R93 ?